src/main/java/org/cbioportal/cdd/service/internal/ClinicalDataDictionaryServiceImpl.java

```diff
@@ -113,6 +114,52 @@
         return clinicalAttributes;
     }
 
+    @Override
+    public List<ClinicalAttributeMetadata> getMetadataBySearchTerms(List<String> searchTerms, String attributeType, boolean inclusiveSearch)
+        throws ClinicalAttributeNotFoundException, ClinicalMetadataSourceUnresponsiveException {
+        assertCacheIsValid();
+        Map<ClinicalAttributeMetadata, Integer> clinicalAttributesToLevenshteinDistance = new HashMap<ClinicalAttributeMetadata, Integer>();
+        List<ClinicalAttributeMetadata> defaultClinicalAttributeCache = new ArrayList<ClinicalAttributeMetadata>(clinicalAttributesCache.getClinicalAttributeMetadata().values());
+
+        // check each attribute for match - skip if doesn't match specified attribute type OR inclusive search that doesn't have all search terms present
+        for (ClinicalAttributeMetadata clinicalAttributeMetadata : defaultClinicalAttributeCache) {
+            if (!clinicalAttributeMetadata.matchesAttributeType(attributeType)) {
```

Comment:
@averyniceday do we have a method that returns all attributes of type `X`? If not then we should make one or use it if it already exists.

Reply:
@averyniceday I started thinking that maybe attributeType should be optional

Reply:
it's optional - if it's null the method will automatically return true